middleware.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 Twitch Interactive, Inc.  All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may not
+// use this file except in compliance with the License. A copy of the License is
+// located at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package twirp
+
+import (
+	"context"
+)
+
+// Method is a method.
+type Method func(ctx context.Context, request interface{}) (interface{}, error)
+
+// Interceptor is a interceptor.
+type Interceptor func(Method) Method
+
+// ChainInterceptors chains the Interceptors.
+//
+// Returns nil if interceptors is empty.
+func ChainInterceptors(interceptors ...Interceptor) Interceptor {
```

Comment:
to be safe, we should remove nils from the array, which will be equivalent to ignoring nil interceptors in the Chain, which makes sense:
```go
intrs := make([]Interceptor, 0, len(interceptors))
for _, intr := range interceptors {
    if intr != nil {
        intrs = append(intrs, intr)
    }
}
```

Then also add a testcase to verify if nil interceptors are properly ignored.

Reply:
This isn't idiomatic Golang to me - I'd expect that users would make sure such inputs are not nil, and would want to be loudly broken if a nil `Interceptor` was put in here. However, I see that `ServerHooks` does this, so I buy the argument to stay consistent here. Updated.